bigchaindb/core.py

```diff
@@ -329,24 +329,32 @@ def get_blocks_status_containing_tx(self, txid):
         else:
             return None
 
-    def get_tx_by_metadata_id(self, metadata_id):
-        """Retrieves transactions related to a metadata.
+    def get_transaction_by_metadata_id(self, metadata_id):
+        """Retrieves valid or undecided transactions related to a particular
+        metadata.
 
-        When creating a transaction one of the optional arguments is the `metadata`. The metadata is a generic
-        dict that contains extra information that can be appended to the transaction.
+        When creating a transaction one of the optional arguments is the
+        `metadata`. The metadata is a generic dict that contains extra
+        information that can be appended to the transaction.
 
-        To make it easy to query the bigchain for that particular metadata we create a UUID for the metadata and
-        store it with the transaction.
+        To make it easy to query the bigchain for that particular metadata we
+        create a UUID for the metadata and store it with the transaction.
 
         Args:
             metadata_id (str): the id for this particular metadata.
 
         Returns:
-            A list of transactions containing that metadata. If no transaction exists with that metadata it
-            returns an empty list `[]`
+            A list of valid or undecided transactions containing that metadata.
+            If no transaction exists with that metadata it returns an empty
+            list `[]`
         """
-        cursor = self.backend.get_transactions_by_metadata_id(metadata_id)
-        return [Transaction.from_dict(tx) for tx in cursor]
+        txids = self.backend.get_txids_by_metadata_id(metadata_id)
+        transactions = []
+        for txid in txids:
+            tx = self.get_transaction(txid)
```

Comment:
maybe a small comment here that `self.get_transaction` returns `None` for invalid `tx's`